src/HotChocolate/Stitching/src/Stitching.Types/Pipeline/ApplyRemove/ApplyRemoveMiddleware.cs

```diff
@@ -0,0 +1,71 @@
+using System.Linq;
+using System.Threading.Tasks;
+using HotChocolate.Language;
+
+namespace HotChocolate.Stitching.Types.Pipeline.ApplyRemove;
+
+/// <summary>
+/// Removes Types and Fields which have the remove directive associated
+/// </summary>
+public sealed class ApplyRemoveMiddleware
+{
+    private readonly RemoveIndexer _indexer = new();
+    private readonly RemoveRewriter _rewriter = new();
+    private readonly MergeSchema _next;
+
+    public ApplyRemoveMiddleware(MergeSchema next)
+    {
+        _next = next;
+    }
+
+    public async ValueTask InvokeAsync(ISchemaMergeContext context)
+    {
+        for (var i = 0; i < context.Documents.Count; i++)
```

Comment:
hmmm ... this might be an issue ... did not really think about this but we do not have the docs grouped in here. I will take a look at the context

Reply:
I thought this would run directly after PrepareDocumentsMiddleware and before ApplyExtensionsMiddleware so that we do have a group of documents?